### PR TITLE
Update google-cloud-cpp to v2.11.0 

### PR DESCRIPTION
**Pull request by dbolduc**
I could not figure out how to add commits onto #138, so here is a new PR.

### CODE REVIEW

**Comment by h-vetinari** [comment]
> I could not figure out how to add commits onto #138, so here is a new PR.

That's because you're not a maintainer on this feedstock... yet? ;-)
I see you're an upstream maintainer, so I'd be happy to have you on board!

(to do so, add your GH handle in the appropriate section at the end of the `meta.yaml`)

**Comment by h-vetinari** [comment]
Also, note that this PR doesn't _automatically_ get the newest protobuf version pulled in (that takes extra steps which are currently waiting in a separate PR). So IIUC, we shouldn't yet remove the patch that disables some features, because this build is still against protobuf 3.21. Some short discussion on this with Carlos is in #136

**Comment by dbolduc** [comment]
> add your GH handle in the appropriate section

Done.

> this PR doesn't automatically get the newest protobuf version pulled in

Ah, ok. I now see how the "Rebuild for protobuf423" PRs update the protobuf version in the `.ci_support/*.yaml` files.

I rebased the patch. (Maybe I could have included the commits from #137 in this PR, but oh well).

**Comment by h-vetinari** [comment]
> Ah, ok. I now see how the "Rebuild for protobuf423" PRs update the protobuf version in the `.ci_support/*.yaml` files.

We don't have to do that manually, that's done by the bot (which can be invoked like `[at]conda-forge-admin, please rerender`, or if you have `conda smithy` installed locally, as `conda smithy rerender` in the feedstock base folder).

If you want we can already pick the relevant commit (https://github.com/conda-forge/google-cloud-cpp-feedstock/pull/137/commits/6da07f47b93a60bec4379fc731c6d11bd8faadfd) into this PR and then rerender. Or we do it in separate PRs (which is perhaps cleaner), no strong feelings.

Happy to have you on board!

File: recipe/patches/0001-disable-libraries.patch [hunk 1]
```diff
@@ -1,17 +1,8 @@
-From 20b43fdb677ea76cfcbdf0e4f000158fe99f615d Mon Sep 17 00:00:00 2001
-From: Carlos O'Ryan <coryan@google.com>
-Date: Tue, 7 Feb 2023 01:35:26 +0000
-Subject: [PATCH] disable libraries
-
----
- cmake/GoogleCloudCppFeatures.cmake | 6 +++---
- 1 file changed, 3 insertions(+), 3 deletions(-)
-
```
Comment (h-vetinari) [inline recipe/patches/0001-disable-libraries.patch:9]: We try to keep attribution on patches. Generally `git format-patch tags/<x> --no-signature` works best (in a branch based on the last release, which can be rebased when the next release comes around).

Comment (h-vetinari) [reply recipe/patches/0001-disable-libraries.patch:9]: I fixed this in #138, also reducing the churn of deleting-and-reinstating the patch.

Comment (dbolduc) [reply recipe/patches/0001-disable-libraries.patch:9]: > I fixed this in https://github.com/conda-forge/google-cloud-cpp-feedstock/pull/138

Awesome, thanks! I will close this PR.

**Pull request closed without merging**